yamlgen/deploy/brupop-resources.yaml

```diff
@@ -0,0 +1,376 @@
+          resources:
+            limits:
+              memory: 50Mi
+            requests:
+              cpu: 10m
+              memory: 50Mi
+          securityContext:
+            seLinuxOptions:
+              level: s0
+              role: system_r
+              type: super_t
+              user: system_u
+          volumeMounts:
+            - mountPath: /run/api.sock
+              name: bottlerocket-api-socket
+            - mountPath: /bin/apiclient
+              name: bottlerocket-apiclient
+            - mountPath: /var/run/secrets/tokens
+              name: bottlerocket-agent-service-account-token
+      imagePullSecrets:
+        - name: brupop
+      serviceAccountName: brupop-agent-service-account
+      volumes:
+        - hostPath:
+            path: /run/api.sock
+            type: Socket
+          name: bottlerocket-api-socket
+        - hostPath:
+            path: /bin/apiclient
+            type: File
+          name: bottlerocket-apiclient
+        - name: bottlerocket-agent-service-account-token
+          projected:
+            sources:
+              - serviceAccountToken:
+                  path: bottlerocket-agent-service-account-token
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    kubernetes.io/service-account.name: brupop-controller-service-account
+  name: brupop-controller-service-account
+  namespace: brupop-bottlerocket-aws
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: brupop-controller-role
+  namespace: brupop-bottlerocket-aws
+rules:
+  - apiGroups:
+      - brupop.bottlerocket.aws
+    resources:
+      - bottlerocketnodes
+      - bottlerocketnodes/status
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - brupop.bottlerocket.aws
+    resources:
+      - bottlerocketnodes
+    verbs:
+      - create
+      - patch
+      - update
+  - apiGroups:
+      - apps
+    resources:
+      - deployments
+    verbs:
+      - create
+      - delete
+      - deletecollection
+      - get
+      - list
+      - patch
+      - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: brupop-controller-role-binding
+  namespace: brupop-bottlerocket-aws
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: brupop-controller-role
+subjects:
+  - kind: ServiceAccount
+    name: brupop-controller-service-account
+    namespace: brupop-bottlerocket-aws
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: brupop-controller
+    app.kubernetes.io/managed-by: brupop
+    app.kubernetes.io/part-of: brupop
+    brupop.bottlerocket.aws/component: brupop-controller
+  name: brupop-controller-deployment
+  namespace: brupop-bottlerocket-aws
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      brupop.bottlerocket.aws/component: brupop-controller
+  strategy:
+    rollingUpdate:
+      maxUnavailable: 100
```

Comment:
Should this be 100%?

Reply:
The controller runs as a single node, to avoid having to coordinate on which nodes to progress next. If it dies, we reschedule it.

I think the right move would be to move away from `rollingUpdate` strategy to [`recreate`](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#recreate-deployment)